### PR TITLE
Add doc for Customizing Pinot

### PR DESCRIPTION
**Pull request by npawar**

### CODE REVIEW

File: docs/customizations.rst [hunk 1]
```diff
@@ -18,5 +18,152 @@
...
+
+New offline data is typically available in a daily or hourly frequency. You can schedule your jobs to run periodically using either cron or a scheduler such as `Azkaban <https://azkaban.github.io/>`_.    
+
+
+2. Pluggable storage
+^^^^^^^^^^^^^^^^^^^^
+We expect the storage to be shared across controllers of the same cluster, such as NFS. You can write your own implementation of PinotFS to store segments in a data layer of your choice, for example Azure or S3. Please refer to `this doc <https://pinot.readthedocs.io/en/latest/pluggable_storage.html>`_ for more details.
```
Comment (sunithabeeram) [inline docs/customizations.rst]: You can use simple references using :ref:<refname> syntax. See other rsts (like code_modules.rst) for example.

File: docs/customizations.rst [hunk 2]
```diff
@@ -18,5 +18,152 @@
...
+
+
+13. Deployables 
+^^^^^^^^^^^^^^^
```
Comment (sunithabeeram) [inline docs/customizations.rst]: Should this be part of admin guide?

File: docs/customizations.rst [hunk 3]
```diff
@@ -18,5 +18,152 @@
...
+
+
+.. image:: img/CustomizingPinot.png
+
+
+1. Generating Pinot segments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Typically, data files will be available on some offline data storage, such as HDFS, and a Hadoop job can be written to read the data and create the segment. The `SegmentCreationJob <https://github.com/apache/incubator-pinot/blob/master/pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentCreationJob.java>`_ class contains a hadoop job for creating segments. This is a map only job, and the mapper can be found in `SegmentCreationMapper <https://github.com/apache/incubator-pinot/blob/master/pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mapper/SegmentCreationMapper.java>`_. You can override the SegmentCreationMapper with a custom mapper by overriding the SegmentCreationJob::getMapperClass() method. 
+
+New offline data is typically available in a daily or hourly frequency. You can schedule your jobs to run periodically using either cron or a scheduler such as `Azkaban <https://azkaban.github.io/>`_.
```
Comment (sunithabeeram) [inline docs/customizations.rst]: We cannot comment on how frequently offline data is available. You can just say that the jobs can be run daily or hourly to push offline data to Pinot.

File: docs/customizations.rst (hunk 2 above)
Comment (sunithabeeram) [reply docs/customizations.rst]: Sorry, my bad. This IS part of admin guide :)

File: docs/customizations.rst (hunk 1 above)
Comment (npawar) [reply docs/customizations.rst]: Fixed

**Comment by npawar** [comment]
Based on comments and discussion, this would be better split out into multiple sections under Extensions. Will do that separately.

**Pull request merged**